reckless_github/src/lib.rs

```diff
@@ -24,10 +25,13 @@ mod tests {
     async fn repository_is_initialized_ok() {
         init();
         let name = "hello";
-        let url = "https://github.com/lightningd/plugins";
-        let mut repo = Github::new(name, url);
+        let url = URL::new("https://github.com/lightningd/plugins");
+        let mut repo = Github::new(name, &url);
         let repo = repo.init().await;
         assert!(repo.is_ok());
-        assert_eq!(Path::new(&(get_dir_path_from_url(url))).exists(), true);
+        assert_eq!(
+            Path::new(&(get_dir_path_from_url(&url.url_string))).exists(),
```

Comment:
```suggestion
            Path::new(&(get_dir_path_from_url(&url.url_string))).exists(),
```
```suggestion
            Path::new(&url.path_string)).exists(),
```

I do not know if it is the same but we can implement the method  `get_dir_path_from_url` inside the URL?

Reply:
maybe a good place to put a FIXME?

Reply:
Fixed. thanks.